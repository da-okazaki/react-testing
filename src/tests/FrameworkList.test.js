import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import FrameworkList from '../FrameworkList';

afterEach(() => cleanup());

const dummyData = [
  {
    id: 1,
    item: 'React',
  },
  {
    id: 2,
    item: 'Angular',
  },
  {
    id: 3,
    item: 'Vue',
  },
];

// test suites
describe('Rendering the list with props', () => {
  // test block
  test('Should render no data ! when no data propped', () => {
    render(<FrameworkList />);
    expect(screen.getByText('No data !')).toBeInTheDocument();
  });
  test('Should render list item correctly', () => {
    render(<FrameworkList frameworks={dummyData} />);
    const frameworkItems = screen.getAllByRole('listitem').map((ele) => ele.textContent);
    const dummyItems = dummyData.map((ele) => ele.item);
    expect(frameworkItems).toEqual(dummyItems);
    expect(screen.queryByText('No Data !')).toBeNull();
  });
  test('Should render snapshot', () => {
    const data = [
      {
        id: 1,
        item: 'React',
      },
      {
        id: 2,
        item: 'Angular',
      },
      {
        id: 3,
        item: 'Vue',
      },
    ];
    const tree = renderer.create(<FrameworkList frameworks={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
