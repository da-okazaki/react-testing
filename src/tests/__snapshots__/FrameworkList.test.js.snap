// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering the list with props Should render snapshot 1`] = `
<div>
  <ul>
    <li>
      React
    </li>
    <li>
      Angular
    </li>
    <li>
      Vue
    </li>
  </ul>
</div>
`;
